swabbie-core/src/test/kotlin/com/netflix/spinnaker/swabbie/ResourceTypeHandlerTest.kt

```diff
@@ -652,84 +650,73 @@ object ResourceTypeHandlerTest {
   }
 
   @Test
-  fun `should notify if required`() {
+  fun `should schedule notification`() {
+    val now = clock.millis()
     val configuration = WorkConfigurationTestHelper.generateWorkConfiguration(
-      notificationConfiguration = NotificationConfiguration(enabled = true, required = true)
-    )
-
-    whenever(dynamicConfigService.getConfig(any(), any(), eq(configuration.maxItemsProcessedPerCycle))) doReturn
-      configuration.maxItemsProcessedPerCycle
-    whenever(ownerResolver.resolve(defaultResource)) doReturn "lucious-mayweather@netflix.com"
-    defaultHandler.setCandidates(mutableListOf(defaultResource))
-    defaultHandler.setRules(defaultRules)
-    whenever(resourceRepository.getMarkedResources()) doReturn listOf(
-      MarkedResource(
-        resource = defaultResource,
-        summaries = listOf(Summary("invalid resource random", "rule 5")),
-        namespace = configuration.namespace,
-        resourceOwner = "test@netflix.com",
-        projectedDeletionStamp = clock.millis().plus(TimeUnit.DAYS.toMillis(3)),
-        markTs = clock.millis()
-      )
+      notificationConfiguration = NotificationConfiguration(enabled = true)
     )
 
-    val updatedMarkedResource = MarkedResource(
+    val markedResource = MarkedResource(
       resource = defaultResource,
       summaries = listOf(Summary("invalid resource random", "rule 5")),
       namespace = configuration.namespace,
-      resourceOwner = "test@netflix.com",
-      projectedDeletionStamp = clock.millis().plus(TimeUnit.DAYS.toMillis(3)),
-      notificationInfo = NotificationInfo(
-        recipient = "test@netflix.com",
-        notificationType = EMAIL.name,
-        notificationStamp = clock.millis()
-      ),
-      markTs = clock.millis()
+      projectedDeletionStamp = now.plus(TimeUnit.DAYS.toMillis(3)),
+      markTs = now
     )
 
+    whenever(
+      dynamicConfigService.getConfig(any(), any(), eq(configuration.maxItemsProcessedPerCycle))
+    ) doReturn configuration.maxItemsProcessedPerCycle
+
+    whenever(
+      resourceRepository.getMarkedResources()
+    ) doReturn listOf(markedResource)
+
+    whenever(
+      notifier.notify(any())
+    ) doReturn NotificationResult(markedResource.resourceOwner, EMAIL, success = true)
+
+    defaultHandler.setCandidates(mutableListOf(defaultResource))
+    defaultHandler.setRules(defaultRules)
+
     defaultHandler.notify(configuration)
-    verify(notifier, times(1)).notify(any(), any(), any())
```

Comment:
Do you want to verify or assert anything in this test? FYI we're using Strikt in `keel` (http://strikt.io/) and it's really good for asserting simple things. I think I already added it to some of the swabbie tests.

Reply:
I will take a look. `Strikt` is used in a few places including in `NotificationSenderTest`